meta-resin-common/recipes-support/resin-hostname/resin-hostname/resin-hostname

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+set -e
+
+. /usr/sbin/resin-vars
+
+if [ -z "$NEW_HOSTNAME" ]; then
+    NEW_HOSTNAME=$(jq -r '.uuid //empty' $CONFIG_PATH)
+fi
+
+if [ -z "$NEW_HOSTNAME" ]; then
+    echo "[ERROR]: Unexpected uuid missing from config.json."
+    exit 1
+fi
+
+# Set hostname temporary hostname
+hostnamectl set-hostname "$NEW_HOSTNAME"
```

Comment:
@telphan I suspect systemd will create this file and using hostnamectl without any flags will update all static, pretty and transient. So hostnamectl will attempt to set pretty in machine-info. Let's just use `--static` as this is the only one we are concerned about for now.